torch/functional.py

```diff
@@ -1438,6 +1438,10 @@ def chain_matmul(*matrices):
     needs to be greater than or equal to 2; if equal to 2 then a trivial matrix-matrix product is returned.
     If :math:`N` is 1, then this is a no-op - the original matrix is returned as is.
 
+    .. note::
+        :func:`torch.chain_matmul` is deprecated, use :func:`torch.linalg.multi_dot` instead. The differences are
```

Comment:
Let's make this a warning and not explain the differences. The one case where a user of chain_matmul() would be frustrated trying to switch to multi_dot() is if they're calling it with a single tensor, which seems incredibly unlikely and is easy to discover.